bin/populate-dependency-track

```diff
@@ -20,74 +20,91 @@ ARGUMENTS
-        fi
+  local repo_name="${1:?repo_name required}"
+  local branch_name="${2}"
+  log 'DEBUG' "cloning ${repo_name}"
+
+  if command -v gh >/dev/null 2>&1; then
+    if [[ -z "${branch_name}" ]]; then
+      gh repo clone companieshouse/"${repo_name}" "${local_dir}"/"${repo_name}"
     else
-        (
-            cd -- "${local_dir}" || exit 1
-
-            if [[ -z "${branch_name}" ]]; then
-                git clone git@github.com:companieshouse/"${repo_name}".git
-            else
-                git clone -b "${branch_name}" git@github.com:companieshouse/"${repo_name}".git
-            fi
-        )
+      gh repo clone companieshouse/"${repo_name}" "${local_dir}"/"${repo_name}" -- -b "${branch_name}"
     fi
+  else
+    (
+      cd -- "${local_dir}" || exit 1
+
+      if [[ -z "${branch_name}" ]]; then
+        git clone git@github.com:companieshouse/"${repo_name}".git
+      else
+        git clone -b "${branch_name}" git@github.com:companieshouse/"${repo_name}".git
+      fi
+    )
+  fi
 }
 
 # Navigates to the supplied repository with name and tries to determine the
 # version either from Github release or a tag or failing that the value in the
 # version file
 function resolve_version() {
-    local repository_name="$1"
-    (
-        cd -- "${local_dir}"/"${repository_name}" || exit 1
+  local repository_name="$1"
+  (
+    cd -- "${local_dir}"/"${repository_name}" || exit 1
 
-        gh release view --json name --jq .name || git describe --tags --abbrev=0 || cat version
-    )
+    gh release view --json name --jq .name || git describe --tags --abbrev=0 || cat version
+  )
 }
 
 # Sets the paraneters for the Concourse task as environment variables -
 # fly will take the values from the environment variables
 function set_parameters() {
-    DEP_TRACK_SERVER='((secrets.dependency-track-server))'
-    DEP_TRACK_API_KEY='((secrets.dependency-track-api-key))'
-    OTHER_CDXGEN_ENV_VARS=""
-    DEP_TRACK_PROJ_TAGS="auto-populated;upload-$(date -u +'%Y-%m-%dT%H:%M:%SZ')"
-    SEND_TO_DEP_TRACK="true"
-    OTHER_CDXGEN_ENV_VARS='ANOTHER_ENV_VAR=yes'
-
-    export DEP_TRACK_SERVER DEP_TRACK_API_KEY SBOM_NAME SPEC_VER
-    export DEP_TRACK_PROJ_VER DEP_TRACK_PROJ_TAGS OTHER_CDXGEN_ENV_VARS SEND_TO_DEP_TRACK
+  DEP_TRACK_SERVER='((secrets.dependency-track-server))'
+  DEP_TRACK_API_KEY='((secrets.dependency-track-api-key))'
+  OTHER_CDXGEN_ENV_VARS=""
+  DEP_TRACK_PROJ_TAGS="auto-populated;upload-$(date -u +'%Y-%m-%dT%H:%M:%SZ')"
+  SEND_TO_DEP_TRACK="true"
+  OTHER_CDXGEN_ENV_VARS='ANOTHER_ENV_VAR=yes'
+
+  export DEP_TRACK_SERVER DEP_TRACK_API_KEY SBOM_NAME SPEC_VER
+  export DEP_TRACK_PROJ_VER DEP_TRACK_PROJ_TAGS OTHER_CDXGEN_ENV_VARS SEND_TO_DEP_TRACK
+}
+
+# attempts to run a no-op task to ensure target is correct
+function test_target_is_correct() {
+  local return_status=0
+
+  log "DEBUG" 'Testing connection to concourse'
+
+  fly -t "${fly_target}" execute \
+    -c "${noop_concourse_task}" >/dev/null || return_status=$?
```

Comment:
1 possible error is the version of fly which could mictmatch. This goes to stderr not /dev/null. Is that a watned behaviour ?
![Screenshot 2024-06-14 at 14 01 39](https://github.com/companieshouse/dependency-track-aws-terraform/assets/12054005/1348a025-f603-4866-8ce3-fe9adea59c39)

Reply:
I guess you should be able to see the errors raised - so I think this is ok? Unless they shouldn't be shown to the user